ui/src/pages/Questions/Detail/components/Reactions/index.tsx

```diff
@@ -0,0 +1,155 @@
+import { FC, memo, useEffect, useState } from 'react';
+import { Button, OverlayTrigger, Tooltip } from 'react-bootstrap';
+import { useTranslation } from 'react-i18next';
+
+import { Icon } from '@/components';
+import { queryReactions, updateReaction } from '@/services';
+import { tryNormalLogged } from '@/utils/guard';
+import { loggedUserInfoStore } from '@/stores';
+
+interface Props {
+  objectId: string;
+  showAddCommentBtn?: boolean;
+  handleClickComment: () => void;
+}
+
+const emojiMap = [
+  {
+    name: 'heart',
+    icon: 'heart-fill',
+    className: 'text-danger',
+  },
+  {
+    name: 'smile',
+    icon: 'emoji-laughing-fill',
+    className: 'text-warning',
+  },
+  {
+    name: 'frown',
+    icon: 'emoji-frown-fill',
+    className: 'text-warning',
+  },
+];
+
+const Index: FC<Props> = ({
+  objectId,
+  showAddCommentBtn,
+  handleClickComment,
+}) => {
+  const [reactions, setReactions] = useState<Record<string, string[]>>({});
+  const { t } = useTranslation('translation');
+  const { username = '' } = loggedUserInfoStore((state) => state.user);
+
+  useEffect(() => {
+    queryReactions(objectId).then((res) => {
+      setReactions(res);
+    });
+  }, []);
+
+  const handleSubmit = (params: { object_id: string; emoji: string }) => {
+    if (!tryNormalLogged(true)) {
+      return;
+    }
+    let reaction = 'activate';
+    if (
+      reactions &&
+      reactions[params.emoji] &&
+      reactions[params.emoji].includes(username)
+    ) {
+      reaction = 'deactivate';
+    }
+    updateReaction({ ...params, reaction }).then((res) => {
+      setReactions(res);
+    });
+  };
+
+  const convertToTooltip = (names: string[]) => {
+    const n: number = Math.min(5, names.length);
+    let ret = '';
+    for (let i = 0; i < n; i += 1) {
+      if (i === n - 1) {
+        ret += names[i];
+      } else {
+        ret += `${names[i]}, `;
+      }
+    }
+    if (names.length > 5) {
+      ret += t('reaction.tooltip', { count: names.length - 5 });
+    }
+    return ret;
+  };
+
+  const renderTooltip = (props) => (
+    <Tooltip id="reaction-button-tooltip" {...props} bsPrefix="tooltip">
+      <div className="d-block d-md-flex flex-wrap m-0 p-0">
+        {emojiMap.map((d) => (
+          <Button
+            key={d.icon}
+            variant="light"
+            size="sm"
+            onClick={() =>
+              handleSubmit({ object_id: objectId, emoji: d.name })
+            }>
+            <Icon name={d.icon} className={d.className} />
+          </Button>
+        ))}
+      </div>
+    </Tooltip>
+  );
+
+  return (
+    <div className="d-block d-md-flex flex-wrap mt-4 mb-3">
```

Comment:
delete `.mt-4` class

Reply:
Hi @shuashuai , when deleting `mb-3`, the reactions will be on top of comments.
![image](https://github.com/apache/incubator-answer/assets/11908658/0ba69ea3-bae2-4cd9-8b74-3fb48c74bb26)

Do you mean by this?
```js
    <div
      className={classNames('d-block d-md-flex flex-wrap', {
        'mb-3': !showAddCommentBtn,
      })}>
```

Reply:
Yes, it should be like this